inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/HeartbeatServiceImpl.java

```diff
@@ -69,27 +71,36 @@ public class HeartbeatServiceImpl implements HeartbeatService {
     private StreamHeartbeatEntityMapper streamHeartbeatMapper;
 
     @Override
-    public Boolean reportHeartbeat(HeartbeatReportRequest request) {
-        if (request == null || StringUtils.isBlank(request.getComponentType())) {
-            log.warn("request is null or component null, just return");
+    public Boolean reportHeartbeat(List<HeartbeatReportRequest> requests) {
+        if (requests == null || requests.isEmpty()) {
             return false;
         }
-        if (log.isDebugEnabled()) {
-            log.debug("received heartbeat: " + request);
-        }
-        heartbeatManager.reportHeartbeat(request);
-        ComponentTypeEnum componentType = ComponentTypeEnum.forName(request.getComponentType());
-        switch (componentType) {
-            case Sort:
-            case DataProxy:
-            case Agent:
-            case Cache:
-            case SDK:
-                return updateHeartbeatOpt(request);
-            default:
-                log.error("Unsupported componentType={} for Inlong", componentType);
-                return false;
+        List<HeartbeatMsg> heartbeatMsgs = new ArrayList<>();
+        for (HeartbeatReportRequest request : requests) {
+            if (request == null || StringUtils.isBlank(request.getComponentType())) {
+                log.warn("request is null or component null, just return");
+                continue;
+            }
+            if (log.isDebugEnabled()) {
+                log.debug("received heartbeat: " + request);
+            }
+            heartbeatMsgs.add(request);
+            ComponentTypeEnum componentType = ComponentTypeEnum.forName(request.getComponentType());
+            switch (componentType) {
+                case Sort:
+                case DataProxy:
+                case Agent:
+                case Cache:
+                case SDK:
```

Comment:
why `updateHeartbeatOpt` only support SDK?

Reply:
Has been rolled back to previous logic.